src/workers.jl

```diff
@@ -120,7 +120,23 @@ function trigger_profile(w::Worker, timeout_profile_wait, from::Symbol=:manual)
         elseif Sys.isbsd()
             kill(w.process, 29)  # SIGINFO
         end
-        sleep(timeout_profile_wait) # Leave time for it to print the profile.
+        sleep(timeout_profile_wait) # Leave time to output the profile.
+    end
+    return nothing
+end
+
+# Spawn GDB to request thread backtraces followed by invoking jl_print_task_backtraces
+# to get task backtraces. We wait for GDB to finish so no sleep is needed. Thread backtraces
+# are written to `gdb.btall`, which is overwritten if present.
+function trigger_backtraces(w::Worker, from::Symbol=:manual)
+    if Sys.islinux()
+        @debug "using GDB to get thread and task backtraces on worker $(w.pid) from $from"
+        gdb_cmd = `gdb -ex "handle SIGSEGV noprint nostop pass" -ex "handle SIGUSR2 noprint nostop pass" -ex "set pagination 0" -ex "set logging overwrite on" -ex "set logging file gdb.btall" -ex "set logging redirect on" -ex "set logging enabled on" -ex "thread apply all bt" -ex "set logging enabled off" -ex "call jl_print_task_backtraces(1)" --batch -p $(w.pid)`
+        try
+            run(gdb_cmd; wait = true) # if `wait = false`, std* are redirected to devnull
+        catch e
+            @warn "Could not get thread/task backtraces via GDB.\n$(sprint(showerror, e))"
```

Comment:
i think the `exception=...` keyword does the right thing (i.e. calls `showerror`) -- is there a reason to prefer interpolating into the string directly?

```suggestion
            @warn "Could not get thread/task backtraces via GDB." exception=e
```